test/module/irohad/consensus/yac/yac_gate_test.cpp

```diff
@@ -208,3 +208,55 @@ TEST_F(YacGateTest, LoadBlockWhenDifferentCommit) {
 
   ASSERT_TRUE(gate_wrapper.validate());
 }
+/**
+ * @given yac gate
+ * @when recives new commit different to the one it voted for
+ * @then polls nodes for the block with corresponding hash untill it succeed
+ */
+TEST_F(YacGateTest, LoadBlockWhenDifferentCommitFailFirst) {
+  // Vote for block => receive different block => load committed block
```

Comment:
This comment should not be required, because there is given, when, then description.